src/Simplify.cpp

```diff
@@ -1449,7 +1449,9 @@ class Simplify : public IRMutator {
         } else if (is_one(b)) {
             expr = a;
         } else if (equal(a, b) &&
-                   !is_zero(b)) {
+                   !is_zero(b) &&
+                   // (n/0)/(n/0) should not be simplified to 1
+                   !(div_a && is_zero(div_a->b))) {
```

Comment:
Ah, ok, I get it -- `signed_integer_overflow` is the sentinel value, not the detection code. Yeah, that makes sense. Let me experiment.